src/reducers/dependencies.reducer.test.js

```diff
@@ -0,0 +1,287 @@
+import {
+  LOAD_DEPENDENCY_INFO_FROM_DISK,
+  ADD_DEPENDENCY_START,
+  ADD_DEPENDENCY_ERROR,
+  ADD_DEPENDENCY_FINISH,
+  UPDATE_DEPENDENCY_START,
+  UPDATE_DEPENDENCY_ERROR,
+  UPDATE_DEPENDENCY_FINISH,
+  DELETE_DEPENDENCY_START,
+  DELETE_DEPENDENCY_ERROR,
+  DELETE_DEPENDENCY_FINISH,
+} from '../actions';
+import { isIterable } from 'core-js';
+
+describe('dependencies reducer', () => {
+  it('should return initial state', () => {
+    expect(reducer(undefined, {})).toEqual({});
+  });
+  it(`should handle ${LOAD_DEPENDENCY_INFO_FROM_DISK}`, () => {
+    const prevState = {
+      baz: {},
+    };
+    const action = {
+      type: LOAD_DEPENDENCY_INFO_FROM_DISK,
+      projectId: 'foo',
+      dependencies: { redux: {} },
+    };
+    expect(reducer(prevState, action)).toMatchSnapshot();
+  });
+
+  it(`should handle ${ADD_DEPENDENCY_START}`, () => {
+    const prevState = {
+      foo: {},
+    };
+
+    const action = {
+      type: ADD_DEPENDENCY_START,
+      projectId: 'foo',
+      dependencyName: 'redux',
+    };
+
+    expect(reducer(prevState, action)).toMatchSnapshot();
+  });
+
+  it(`should handle ${ADD_DEPENDENCY_ERROR}`, () => {
+    const prevState = {
+      foo: {
+        redux: {},
+        'react-router': {},
+      },
+    };
+
+    const action = {
+      type: ADD_DEPENDENCY_ERROR,
+      projectId: 'foo',
+      dependencyName: 'redux',
+    };
+
+    expect(reducer(prevState, action)).toMatchSnapshot();
+  });
+
+  it(`should handle ${ADD_DEPENDENCY_FINISH}`, () => {
+    const prevState = {
+      foo: {
+        redux: {
+          name: 'redux',
+          status: 'installing',
+          description: '',
+          homepage: '',
+          license: '',
+          repository: '',
+          version: '',
+        },
+      },
+    };
+
+    const action = {
+      type: ADD_DEPENDENCY_FINISH,
+      projectId: 'foo',
+      dependency: {
+        name: 'redux',
+        status: 'idle',
+        description: 'dependency description',
+        keywords: ['key', 'words'],
+        version: '3.2',
+        homepage: 'https://dependency-homepage.io',
+        license: 'MIT',
+        repository: 'https://github.com',
+      },
+    };
+
+    expect(reducer(prevState, action)).toMatchSnapshot();
+  });
+
+  it(`should handle ${UPDATE_DEPENDENCY_START}`, () => {
+    const prevState = {
+      foo: {
+        redux: {
+          name: 'redux',
+          status: 'idle',
+          description: 'dependency description',
+          keywords: ['key', 'words'],
+          version: '3.2',
+          homepage: 'https://dependency-homepage.io',
+          license: 'MIT',
+          repository: 'https://github.com',
+        },
+      },
+    };
+
+    const action = {
+      type: UPDATE_DEPENDENCY_START,
+      projectId: 'foo',
+      dependencyName: 'redux',
+    };
+
+    expect(reducer(prevState, action)).toMatchSnapshot();
+  });
+  it(`should handle ${UPDATE_DEPENDENCY_ERROR}`, () => {
```

Comment:
Super-nitpick: Missing a blank line above this one.

Reply:
fixed